utils/s3/client.cc

```diff
@@ -761,7 +761,11 @@ future<> client::multipart_upload::abort_upload() {
     s3l.trace("DELETE upload {}", _upload_id);
     auto req = http::request::make("DELETE", _client->_host, _object_name);
     req.query_parameters["uploadId"] = std::exchange(_upload_id, ""); // now upload_started() returns false
-    co_await _client->make_request(std::move(req), ignore_reply, http::reply::status_type::no_content);
+    co_await _client->make_request(std::move(req), ignore_reply, http::reply::status_type::no_content).handle_exception([](const std::exception_ptr&) -> future<> {
+        // Here we discard whatever exception is thrown when aborting multipart upload since we don't care about cleanly aborting it since there are other means
```

Comment:
I think we should at least report (warning I guess) that there was an exception.

Reply:
warning is too much IMO. we really dont care about it happening. Even `info` is too much, I think
@xemul , @tchaikov What do you think guys?

Reply:
If we don't care at all, we can also pass nullopt as expected error code as well

Reply:
Do we have any way to collect statistics about failures?
If we do, then no need to log.
If logging is our only way at the moment, then at least for now I think it would be great to have it.
We can remove this logging later if we find it is irritating.

Reply:
> warning is too much IMO. we really dont care about it happening. Even `info` is too much, I think @xemul , @tchaikov What do you think guys?

the thing is how we will do with this failures. per https://docs.aws.amazon.com/AmazonS3/latest/userguide/abort-mpu.html, Amazon charges for the uploaded parts. so i think we should at least aware of these unexpected error. we could use these logging messages as one of the supporting material when we face an unexpected bill from Amazon?

Reply:
> > warning is too much IMO. we really dont care about it happening. Even `info` is too much, I think @xemul , @tchaikov What do you think guys?
> 
> the thing is what to do with these failures. per https://docs.aws.amazon.com/AmazonS3/latest/userguide/abort-mpu.html, Amazon charges for the uploaded parts. so i think we should at least aware of these unexpected errors. we could use these logging messages as one of the supporting material when we face an unexpected bill from Amazon?

https://aws.amazon.com/blogs/aws-cloud-financial-management/discovering-and-deleting-incomplete-multipart-uploads-to-lower-amazon-s3-costs/

Reply:
> Do we have any way to collect statistics about failures? If we do, then no need to log. If logging is our only way at the moment, then at least for now I think it would be great to have it. We can remove this logging later if we find it is irritating.

logging added